bokehjs/src/lib/models/widgets/spinner.ts

```diff
@@ -201,13 +187,21 @@ export class SpinnerView extends NumericInputView {
       else if (step < 0)
         this.model.value = (high!=null)? high : (low!=null)? max(low, 0) : 0
     } else
-      this.model.value = this.adjust_to_precision(this.model.value + step)
+      this.model.value = this.bound_value(this.model.value + step)
   }
 
   override change_input(): void {
     super.change_input()
     this.model.value_throttled = this.model.value
   }
+
+  override bound_value(value: number): number {
+    let output = value
+    const {low, high} = this.model
+    if (low != null && output < low) output = this.model.value ?? 0
+    if (high != null && output > high) output = this.model.value ?? 0
+    return output
```

Comment:
We try to avoid `let` in the codebase wherever possible, suggest:
```suggestion
    const {low, high, value} = this.model
    if (low != null && output < low) return value ?? 0
    if (high != null && output > high) return value ?? 0
    return value
```